forge/test/mlir/llama/test_llama_inference.py

```diff
@@ -126,3 +127,141 @@ def test_llama_inference_cache_cpu():
+    # untill the a specified maximum number of new tokens is reached or an end-of-sequence token is encountered.
+    max_new_tokens = 32
+    for _ in range(max_new_tokens):
+
+        # CPU Inference
+        framework_output = framework_model(input_ids)
+
+        # Compile the model
+        tt_inputs = [input_ids]
+        compiled_model = forge.compile(framework_model, sample_inputs=tt_inputs)
+
+        # Run on TT device
+        tt_output = compiled_model(*tt_inputs)
+        tt_output = [tt_out.to("cpu") for tt_out in tt_output]
+
+        # Validate TT result with Framework
+        assert all([compare_with_golden_pcc(golden=fw_out, calculated=tt_out, pcc=0.99) for fw_out, tt_out in zip(framework_output, tt_output)])
+        next_token_logits = tt_output[0][:, -1, :]
+        next_token = torch.argmax(next_token_logits, dim=-1)
+
+        if next_token == tokenizer.eos_token_id:
+            break
+
+        input_ids = torch.cat(
+            [input_ids, next_token.unsqueeze(0)], dim=-1
+        )
+
+
+    # Generated text
+    generated_text = tokenizer.decode(input_ids[0], skip_special_tokens=True)
+    print("generated_text=",generated_text)
+
+
+@pytest.mark.xfail()
+def test_llama_inference_cache_tt():
+
+    # Load Llama 3B model and tokenizer
+    model_path = "openlm-research/open_llama_3b"
+    framework_model = load_model(model_path, use_cache=True)
+    tokenizer = LlamaTokenizer.from_pretrained(model_path)
+
+    class Llama_Model_Wrapper(torch.nn.Module):
+        def __init__(self, model):
+            super().__init__()
+            self.model = model
+        def forward(self, inputs):
+            past_key_values_list = inputs[1:]
+            past_key_values = list()
+            for idx in range(0, len(past_key_values_list), 2):
+                past_key_values.append(tuple([past_key_values_list[idx], past_key_values_list[idx + 1]]))
+            model_outputs = self.model(input_ids=inputs[0], past_key_values=tuple(past_key_values))
+            outputs = [model_outputs[0]]
+            for k, v in model_outputs[1]:
+                outputs.append(k)
+                outputs.append(v)
+            return outputs
+
+    llama_model= Llama_Model_Wrapper(framework_model)
+    llama_model.eval()
+
+    # Prepare input sentence
+    prompt = "Q: What is the largest animal?\nA:"
+    inputs = tokenizer(prompt, return_tensors="pt")
+
+    # Run Prefill on CPU with cache to get the initial logits and past key-values
+    logits, past_key_values = framework_model(input_ids=inputs.input_ids, attention_mask=inputs.attention_mask)
+    next_token_logits = logits[:, -1, :]
+    next_token = torch.argmax(next_token_logits, dim=-1)
+
+    generated_tokens = inputs.input_ids
+    generated_tokens = torch.cat(
+        [generated_tokens, next_token.unsqueeze(0)], dim=-1
+    )
+
+    model_inputs = [next_token.unsqueeze(0)]
```

Comment:
One general question, do we need to bundle the KV cache with the next token in the same input? 

Can we have single input that will be next token, and another that will represent the past cache?

Reply:
Unbundled the input token and past key values to make it as two separate inputs to the wrapper forward function